webdriver-spec.html

```diff
@@ -1435,288 +1435,506 @@
 
-   <dt>"<code>browserVersion</code>"
-   <dd>Version of the user agent
+<table class=simple>
+<thead>
+<tr>
+<th>Key
+<th>Value Type
+<th>Description
+</tr>
+</thead>
+
+<tbody>
+<tr>
+<td>"<code>implicit</code>"
+<td>number
+<td>Specifies the <a>session implicit wait timeout</a>.
+</tr>
+
+<tr>
+<td>"<code>page load</code>"
+<td>number
+<td>Specifies the <a>session page load timeout</a>.
+</tr>
+
+<tr>
+<td>"<code>script</code>"
+<td>number
+<td>Specifies the <a>session script timeout</a>.
+</tr>
+</table>
 
-   <dt>"<code>platformName</code>"
-   <dd>Lowercase name of the platform
+</section><!-- /Session Timeouts Configuration -->
 
-   <dt>"<code>platformVersion</code>"
-   <dd>Version of the platform
+<section>
+<h3>Processing Capabilities</h3>
 
-   <dt>"<code>acceptSslCerts</code>"
-   <dd>False
-  </dl>
+<p>When <dfn>processing capabilities</dfn> with argument <var>parameters</var>,
+  the <a>endpoint node</a> <em>MUST</em> run the following steps:
 
- <li><p>Let <var>required capabilities</var> be the result of <a>getting a property</a>
-  named <code>requiredCapabilities</code> from <var>capabilities</var>.
+<ol>
+<li><p>Let <var>capabilities request</var> be the result of getting a property
+named "<code>capabilities</code>" from <var>parameters</var>.
 
- <li>If <var>required capabilities</var> is not a JSON Object,
-  set the value to an empty JSON Object.
+  <ol>
+  <li><p>If <var>capabilities request</var> is not a JSON object,
+  return <a>error</a> with <a>error code</a> <a>invalid argument</a>.
+  </ol>
 
- <li><p>Let <var>desired capabilities</var> be the result of <a>getting a property</a>
-  named <code>desiredCapabilities</code> from <var>capabilities</var>.
+<li><p>Let <var>required capabilities</var> be the result of getting a property
+named "<code>alwaysMatch</code>" from <var>capabilities request</var>.
 
- <li>If <var>desired capabilities</var> is not a JSON Object,
+  <ol>
+  <li><p>If <var>required capabilities</var> is <a>null</a> or <a>undefined</a>,
   set the value to an empty JSON Object.
 
- <li><p>Let <var>length</var> be the length of <var>required capabilities</var>.
-
- <li><p>Let <var>k</var> be 0.
+  <li><p>If <var>required capabilities</var> is not a JSON Object,
+  return <a>error</a> with <a>error code</a> <a>invalid argument</a>.
+  </ol>
 
- <li><p>While <var>k</var> &lt; <var>length</var>:
+<li><p>Let <var>desired capabilities</var> be the result of getting a property
+named "<code>firstMatch</code>" from <var>capabilities request</var>.
 
   <ol>
-   <li><p>Let <var>capability</var> be the entry
-    in <var>required capabilities</var> at index <var>k</var>.
-
-   <li><p>If the name of the <var>capability</var> is among
-    the names of entries in <var>desired capabilities</var>,
-    remove the corresponding entry from <var>desired capabilities</var>.
+  <li><p>If <var>desired capabilities</var> is <a>null</a> or <a>undefined</a>,
+  set the value to an empty JSON List.
```

Comment:
This should set the value to `[{}]`, rather than `[]`, so that there is at least one candidate capability set to consider.